aws/resource_aws_lb_listener.go

```diff
@@ -151,11 +155,26 @@ func resourceAwsLbListenerCreate(d *schema.ResourceData, meta interface{}) error
 func resourceAwsLbListenerRead(d *schema.ResourceData, meta interface{}) error {
 	elbconn := meta.(*AWSClient).elbv2conn
 
-	resp, err := elbconn.DescribeListeners(&elbv2.DescribeListenersInput{
+	var resp *elbv2.DescribeListenersOutput
+	var request = &elbv2.DescribeListenersInput{
 		ListenerArns: []*string{aws.String(d.Id())},
+	}
+
+	err := resource.Retry(d.Timeout(schema.TimeoutRead), func() *resource.RetryError {
```

Comment:
When dealing with AWS service eventual consistency, we prefer to hardcode 1 minute timeouts. Generally its not desired to allow that timeout to be user configurable as its a knob that is unlikely to handle the situation properly by waiting longer.